fix: shutdown app in node_t dtor

shaitan opened the pull request:
Shutdown signal is asynchronously invoked, so it can be handled by node service when context is already destroyed.

karitra commented:
It seems that `apps->clear()` would be called in any way in destructor and what is the reason to keep `signal` instance in proposed scheme without any binded callback?

shaitan commented:
You are right, PTAL at updated PR - I removed explicit `app->clear()` and also removed signals from `node_t` since now it doesn't listen and handle any signal.
